packages/@sanity/form-builder/src/inputs/PortableText/Editor.tsx

```diff
@@ -170,13 +87,45 @@ export function Editor(props: EditorProps) {
     }
   }, [isFullscreen, isTopLayer, onToggleFullscreen])
 
-  const handleScrollSelectionIntoView = useMemo(
-    () => createScrollSelectionIntoView(scrollElement),
-    [scrollElement]
+  // Keep the editor focused even though we are clicking on the background or the toolbar of the editor.
+  const keepFocused = useCallback((event: React.SyntheticEvent) => {
+    if (event.target instanceof Node && !editableRef.current.contains(event.target)) {
+      event.preventDefault()
+      event.stopPropagation()
+    }
+  }, [])
+
+  const editable = useMemo(
+    () => (
+      <PortableTextEditable
+        hotkeys={hotkeys}
+        onCopy={onCopy}
+        onPaste={onPaste}
+        placeholderText="Empty"
+        ref={editableRef}
+        renderAnnotation={renderAnnotation}
+        renderBlock={renderBlock}
+        renderChild={renderChild}
+        renderDecorator={renderDecorator}
+        scrollSelectionIntoView={scrollSelectionIntoView}
+        selection={initialSelection}
+        spellCheck={false} // This is taken care of via renderBlock prop
+      />
+    ),
+    [
+      hotkeys,
+      initialSelection,
+      onCopy,
+      onPaste,
+      renderAnnotation,
+      renderBlock,
+      renderChild,
+      scrollSelectionIntoView,
+    ]
```

Comment:
I believe we don’t really need to wrap this in `useMemo`. It doesn’t add much perf improvement, but actually adds another operation instead. Think it can just be rendered in the `return` below.

PS. We should instead wrap the `PortableTextEditable` in a `memo` if we only want it to re-render on a prop change.

```tsx
const MemoPortableTextEditable = memo(PortableTextEditable)

// ...

<MemoPortableTextEditable ... />
```

Reply:
I know we removed the memo in the last iteration, but then `value` and other stuff was part of the deps. The deps now are pretty much non-changing, and that's why I'm thinking a memo makes sense here now.